### ISSUE TITLE
awsx.ec2.Vpc() fails with TypeError: Missing 'subnet_id' argument

### ISSUE THREAD

**Issue by smedegaard** [pulumi/pulumi-awsx#1144]
### What happened?

I’m trying to create a VPC using [awsx](https://www.pulumi.com/registry/packages/awsx/api-docs/ec2/vpc/), using the Python SDK

which fails with

```
  File "/Users/anders/Library/Caches/pypoetry/virtualenvs/infra-1e2fA1oY-py3.10/lib/python3.10/site-packages/pulumi_aws/ec2/nat_gateway.py", line 63, in _configure
        raise TypeError("Missing 'subnet_id' argument")
    TypeError: Missing 'subnet_id' argument
```

Seems like the subnet ids are not pass properly internally in the method?

### Example

```python
import pulumi_awsx as awsx

# Create a VPC
debezium_vpc = awsx.ec2.Vpc("my_fancy_vpc")
```

```bash
Diagnostics:
  pulumi:pulumi:Stack (infra-dev):
    error: Program failed with an unhandled exception:
    Traceback (most recent call last):
      File "/usr/local/bin/pulumi-language-python-exec", line 197, in <module>
        loop.run_until_complete(coro)
      File "/Users/anders/.asdf/installs/python/3.10.3/lib/python3.10/asyncio/base_events.py", line 646, in run_until_complete
        return future.result()
      File "/Users/anders/Library/Caches/pypoetry/virtualenvs/infra-1e2fA1oY-py3.10/lib/python3.10/site-packages/pulumi/runtime/stack.py", line 137, in run_in_stack
        await run_pulumi_func(lambda: Stack(func))
      File "/Users/anders/Library/Caches/pypoetry/virtualenvs/infra-1e2fA1oY-py3.10/lib/python3.10/site-packages/pulumi/runtime/stack.py", line 51, in run_pulumi_func
        await wait_for_rpcs()
      File "/Users/anders/Library/Caches/pypoetry/virtualenvs/infra-1e2fA1oY-py3.10/lib/python3.10/site-packages/pulumi/runtime/stack.py", line 121, in wait_for_rpcs
        raise exception
      File "/Users/anders/Library/Caches/pypoetry/virtualenvs/infra-1e2fA1oY-py3.10/lib/python3.10/site-packages/pulumi/runtime/rpc_manager.py", line 71, in rpc_wrapper
        result = await rpc
      File "/Users/anders/Library/Caches/pypoetry/virtualenvs/infra-1e2fA1oY-py3.10/lib/python3.10/site-packages/pulumi/output.py", line 103, in is_value_known
        return await is_known and not contains_unknowns(await future)
      File "/Users/anders/Library/Caches/pypoetry/virtualenvs/infra-1e2fA1oY-py3.10/lib/python3.10/site-packages/pulumi/output.py", line 103, in is_value_known
        return await is_known and not contains_unknowns(await future)
      File "/Users/anders/Library/Caches/pypoetry/virtualenvs/infra-1e2fA1oY-py3.10/lib/python3.10/site-packages/pulumi/output.py", line 103, in is_value_known
        return await is_known and not contains_unknowns(await future)
      [Previous line repeated 27 more times]
      File "/Users/anders/Library/Caches/pypoetry/virtualenvs/infra-1e2fA1oY-py3.10/lib/python3.10/site-packages/pulumi/runtime/resource.py", line 1029, in do_register
        rpc.resolve_outputs(
      File "/Users/anders/Library/Caches/pypoetry/virtualenvs/infra-1e2fA1oY-py3.10/lib/python3.10/site-packages/pulumi/runtime/rpc.py", line 1104, in resolve_outputs
        for key, value in deserialize_properties(outputs).items():
      File "/Users/anders/Library/Caches/pypoetry/virtualenvs/infra-1e2fA1oY-py3.10/lib/python3.10/site-packages/pulumi/runtime/rpc.py", line 637, in deserialize_properties
        value = deserialize_property(v, keep_unknowns)
      File "/Users/anders/Library/Caches/pypoetry/virtualenvs/infra-1e2fA1oY-py3.10/lib/python3.10/site-packages/pulumi/runtime/rpc.py", line 759, in deserialize_property
        values = [deserialize_property(v, keep_unknowns) for v in value]  # type: ignore
      File "/Users/anders/Library/Caches/pypoetry/virtualenvs/infra-1e2fA1oY-py3.10/lib/python3.10/site-packages/pulumi/runtime/rpc.py", line 759, in <listcomp>
        values = [deserialize_property(v, keep_unknowns) for v in value]  # type: ignore
      File "/Users/anders/Library/Caches/pypoetry/virtualenvs/infra-1e2fA1oY-py3.10/lib/python3.10/site-packages/pulumi/runtime/rpc.py", line 769, in deserialize_property
        props = deserialize_properties(value, keep_unknowns)
      File "/Users/anders/Library/Caches/pypoetry/virtualenvs/infra-1e2fA1oY-py3.10/lib/python3.10/site-packages/pulumi/runtime/rpc.py", line 617, in deserialize_properties
        return deserialize_resource(props_struct, keep_unknowns)
      File "/Users/anders/Library/Caches/pypoetry/virtualenvs/infra-1e2fA1oY-py3.10/lib/python3.10/site-packages/pulumi/runtime/rpc.py", line 670, in deserialize_resource
        return cast("Resource", resource_module.construct(urn_name, typ, urn))
      File "/Users/anders/Library/Caches/pypoetry/virtualenvs/infra-1e2fA1oY-py3.10/lib/python3.10/site-packages/pulumi_aws/_utilities.py", line 210, in construct
        return TheClass(name, pulumi.ResourceOptions(urn=urn))
      File "/Users/anders/Library/Caches/pypoetry/virtualenvs/infra-1e2fA1oY-py3.10/lib/python3.10/site-packages/pulumi_aws/ec2/nat_gateway.py", line 615, in __init__
        NatGatewayArgs._configure(_setter, **kwargs)
      File "/Users/anders/Library/Caches/pypoetry/virtualenvs/infra-1e2fA1oY-py3.10/lib/python3.10/site-packages/pulumi_aws/ec2/nat_gateway.py", line 63, in _configure
        raise TypeError("Missing 'subnet_id' argument")
    TypeError: Missing 'subnet_id' argument
    
 ```

### Output of `pulumi about`

CLI
Version      3.91.1
Go Version   go1.21.3
Go Compiler  gc

Plugins
NAME    VERSION
aws     6.6.1
awsx    2.0.3
docker  4.4.4
python  unknown

Host
OS       darwin
Version  14.0
Arch     x86_64

This project is written in python: executable='/Users/anders/Library/Caches/pypoetry/virtualenvs/infra-1e2fA1oY-py3.10/bin/python3' version='3.10.3'

Current Stack: smedegaard/infra/dev

Found no resources associated with dev

Found no pending operations associated with dev

Backend
Name           pulumi.com
URL            https://app.pulumi.com/smedegaard
User           smedegaard
Organizations  smedegaard
Token type     personal

Dependencies:
NAME         VERSION
pip          22.3.1
pulumi-awsx  2.0.3
setuptools   65.6.3
wheel        0.38.4

Pulumi locates its logs in /var/folders/tl/06ddx4rj5tj_5h3gpvvtxtbm0000gn/T/ by default
(infra-py3.10)

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by mikhailshilkov** [comment]
@smedegaard  I think this has been fixed in AWSX 2.1.0 that was released today and AWS 6.7.0 released two days ago. Could you please give those a try?

**Comment by smedegaard** [comment]
Confirming that it work with updated dependencies

  • Updating grpcio (1.59.0 -> 1.56.2)
  • Updating protobuf (4.24.4 -> 4.25.0)
  • Updating pulumi (3.91.1 -> 3.92.0)
  • Updating pulumi-aws (6.6.1 -> 6.7.0)
  • Updating pulumi-awsx (2.0.3 -> 2.1.1)
